code/game/objects/structures/tables_racks.dm

```diff
@@ -298,25 +298,16 @@
 	framestack = /obj/item/stack/rods
 	buildstack = /obj/item/stack/tile/carpet
 	canSmoothWith = list(/obj/structure/table/wood/fancy, /obj/structure/table/wood/fancy/black)
+	var/smooth_icon = 'icons/obj/smooth_structures/fancy_table.dmi'
 
-/obj/structure/table/wood/fancy/New()
-	// New() is used so that the /black subtype can override `icon` easily and
-	// the correct value will be used by the smoothing subsystem.
+/obj/structure/table/wood/fancy/Initialize()
 	. = ..()
-	// Needs to be set dynamically because table smooth sprites are 32x34,
-	// which the editor treats as a two-tile-tall object. The sprites are that
-	// size so that the north/south corners look nice - examine the detail on
-	// the sprites in the editor to see why.

```

Comment:
I would suggest keeping this comment. Also, is this tested? I have a feeling it doesn't work.

Reply:
I'll add a comment. I tested it some time ago and it should work (can't test right now, trying to get wine to work)